homeassistant/components/blue_current/button.py

```diff
@@ -0,0 +1,101 @@
+"""Support for Blue Current buttons."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from bluecurrent_api.client import Client
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Connector
+from .const import DOMAIN
+from .entity import ChargepointEntity
+
+
+@dataclass(frozen=True)
+class ChargePointButtonEntityDescriptionMixin:
+    """Mixin for the called functions."""
+
+    function: Callable[[Client, str], Coroutine[Any, Any, None]]
+
+
+@dataclass(frozen=True)
+class ChargePointButtonEntityDescription(
+    ButtonEntityDescription, ChargePointButtonEntityDescriptionMixin
+):
+    """Describes a Blue Current button entity."""
```

Comment:
We don't need mixin's anymore.

```suggestion
@dataclass(kw_only=True, frozen=True)
class ChargePointButtonEntityDescription(ButtonEntityDescription):
    """Describes a Blue Current button entity."""

    function: Callable[[Client, str], Coroutine[Any, Any, None]]
```